packages/swap/src/routes/v2/quote.ts

```diff
@@ -0,0 +1,334 @@
+  estimatedBoostFeeBps?: number;
+}) => {
+  const lastChunkRatio = new BigNumber(dcaQuoteParams.lastChunkAmount.toString()).dividedBy(
+    new BigNumber(dcaQuoteParams.chunkSize.toString()),
+  );
+  // eslint-disable-next-line no-param-reassign
+  dcaQuote.type = 'DCA';
+  if (dcaQuoteParams && dcaQuote) {
+    const netWorkFee = dcaQuote.includedFees.find((fee) => fee.type === 'NETWORK');
+    if (netWorkFee) {
+      netWorkFee.amount = new BigNumber(netWorkFee.amount)
+        .multipliedBy(dcaQuoteParams.numberOfChunks - 1)
+        .plus(new BigNumber(netWorkFee.amount).multipliedBy(lastChunkRatio))
+        .toFixed(0);
+    }
+
+    // eslint-disable-next-line no-param-reassign
+    dcaQuote.egressAmount = new BigNumber(dcaQuote.egressAmount)
+      .multipliedBy(dcaQuoteParams.numberOfChunks - 1)
+      .plus(BigNumber(dcaQuote.egressAmount).multipliedBy(lastChunkRatio))
+      .toFixed(0);
+    // eslint-disable-next-line no-param-reassign
+    dcaQuote.estimatedDurationSeconds += dcaQuoteParams.addedDurationSeconds;
+  }
+  if (dcaQuoteParams && dcaBoostedQuote && estimatedBoostFeeBps) {
+    const netWorkFee = dcaBoostedQuote.includedFees.find((fee) => fee.type === 'NETWORK');
+    if (netWorkFee) {
+      netWorkFee.amount = new BigNumber(netWorkFee.amount)
+        .multipliedBy(dcaQuoteParams.numberOfChunks - 1)
+        .plus(BigNumber(netWorkFee.amount).multipliedBy(lastChunkRatio))
+        .toFixed(0);
+    }
+
+    const boostFee = dcaBoostedQuote.includedFees.find((fee) => fee.type === 'BOOST');
+    if (boostFee) {
+      boostFee.amount = new BigNumber(boostFee.amount)
+        .multipliedBy(dcaQuoteParams.numberOfChunks - 1)
+        .plus(BigNumber(boostFee.amount).multipliedBy(lastChunkRatio))
+        .toFixed(0);
+    }
+
+    // eslint-disable-next-line no-param-reassign
+    dcaQuote.boostQuote = {
+      ...dcaBoostedQuote,
+      estimatedBoostFeeBps,
+      egressAmount: BigNumber(dcaBoostedQuote.egressAmount)
+        .multipliedBy(dcaQuoteParams.numberOfChunks - 1)
+        .plus(BigNumber(dcaBoostedQuote.egressAmount).multipliedBy(lastChunkRatio))
+        .toFixed(0),
+      estimatedDurationSeconds:
+        dcaBoostedQuote.estimatedDurationSeconds + dcaQuoteParams.addedDurationSeconds,
+    };
+  }
+};
+
+const fallbackChains = {
+  [Assets.ETH]: Chains.Ethereum,
+  [Assets.USDC]: Chains.Ethereum,
+  [Assets.FLIP]: Chains.Ethereum,
+  [Assets.BTC]: Chains.Bitcoin,
+  [Assets.DOT]: Chains.Polkadot,
+  [Assets.USDT]: Chains.Ethereum,
+  [Assets.SOL]: Chains.Solana,
+} satisfies Record<Asset, Chain>;
+
+const quoteRouter = (io: Server) => {
+  const quoter = new Quoter(io);
+
+  const router = express.Router().use((req, res, next) => {
+    if (env.DISABLE_QUOTING) {
+      next(ServiceError.unavailable('Quoting is currently unavailable due to maintenance'));
+
+      return;
+    }
+
+    next();
+  });
+
+  router.get(
+    '/',
+    asyncHandler(async (req, res) => {
```

Comment:
i think a lot of the code in the handler is duplicated, right? maybe we could reuse the same handler with a query param to enable dca quotes?

Reply:
so if the query param is set we return an array, otherwise we keep the previous behaviour

Reply:
I thought about it, but I feel like it'd be strange if the return type would change so drastically (from object to array of objects) based on the params.
I think it's better to switch to a new v2 endpoint to always return an array of quotes

Reply:
alright fine for me too 🙂 but it would be nice to reuse the code if possible to not diverge - maybe we can extract a method that returns the array and then only stringify the first one in the `v1` handler?